scipy/__init__.py

```diff
@@ -56,7 +56,13 @@
  __numpy_version__ --- Numpy version string
 
 """
-__all__ = ['test']
+
+
+def __dir__():
+    return ['test']
```

Comment:
I did a clean build of this feature branch locally and confirmed that the (i.e., IPython) autocomplete goes away while the deprecated functions remain accessible via full specification--`scipy.arange(5)` (!)

I was initially worried that `dir(scipy)` might have something useful (it is about 600 elements long at the moment), but seems to mostly just be the deprecated NumPy aliases, so let's try this I guess.

Thanks Stefan